ldp/alg/rollout.py

```diff
@@ -42,8 +43,9 @@ def reraise_exc_as(reraise: type[CaughtError], enabled: bool) -> Iterator[None]:
         yield
     except Exception as e:
         if enabled:
-            logger.exception(f"Caught {reraise.exc_type} exception.")
-            raise reraise(e) from e
+            error_details = format_error_details(e)
+            logger.exception(f"Caught {reraise.exc_type} exception:\n{error_details}")
+            raise reraise(e) from None
```

Comment:
Why `from None` instead of `from e`? I might be missing something obvious here

Reply:
Error chaining was adding redundant information and I'm not sure exactly why. Have a look at the traceback with chaining below:

**With chaining (from e)**
```
2024-12-30 20:41:29,924 - ldp.alg.rollout - ERROR - Caught env exception.
Traceback (most recent call last):
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/ldp/alg/rollout.py", line 43, in reraise_exc_as
    yield
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/ldp/alg/rollout.py", line 230, in _rollout
    obs, tools = await env.reset()
                 ^^^^^^^^^^^^^^^^^
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/aviary/env_client.py", line 102, in reset
    return await super().reset()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/aviary/env_client.py", line 46, in reset
    response = await self._post(
               ^^^^^^^^^^^^^^^^^
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/aviary/env_client.py", line 42, in _post
    response.raise_for_status()
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/httpx/_models.py", line 763, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Server error '500 Internal Server Error' for url 'http://localhost:8080/reset'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500
Traceback (most recent call last):               
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/ldp/alg/rollout.py", line 43, in reraise_exc_as
    yield
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/ldp/alg/rollout.py", line 230, in _rollout
    obs, tools = await env.reset()
                 ^^^^^^^^^^^^^^^^^
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/aviary/env_client.py", line 102, in reset
    return await super().reset()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/aviary/env_client.py", line 46, in reset
    response = await self._post(
               ^^^^^^^^^^^^^^^^^
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/aviary/env_client.py", line 42, in _post
    response.raise_for_status()
  File "/Users/ludovicomitchener/Desktop/repos/aviary-internal/.venv/lib/python3.12/site-packages/httpx/_models.py", line 763, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Server error '500 Internal Server Error' for url 'http://localhost:8080/reset'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500
```

It doesn't include the source of the http error and shows the `reraise_exc_as` traceback twice.